### ISSUE TITLE
`kthvalue` triggers segmentation fault

### ISSUE THREAD

**Issue by TestSomething22** [pytorch/pytorch#73628]
### 🐛 Describe the bug

`kthvalue` triggers segmentation fault with some specific value

```python
import torch
input = torch.rand([1, 210120], dtype=torch.float32)
k = 9
dim = 0
torch.kthvalue(input.clone(), k, dim=dim)
```

### Versions

pytorch: 1.10.1

cc @ezyang @gchanan @zou3519

**Comment by emcastillo** [comment]
Seems like in the current master branch now this throws a proper exception.

```
Traceback (most recent call last):
  File "kth.py", line 5, in <module>
    torch.kthvalue(input.clone(), k, dim=dim)
RuntimeError: kthvalue(): selected number k out of range for dimension 0
```

**Comment by ngimel** [comment]
No, we don't backport fixes like this.

**Comment by Jup11ter** [comment]
I have observed similar buggy behavior on `torch.Tensor.kthvalue` in PyTorch 1.9.0 and 1.10.0. The following codes will raise:
```
Segmentation fault (core dumped)
```

However, in 1.11.0 and later versions, it will raise `RuntimeError: kthvalue(): selected number k out of range for dimension 0`, which is expected behavior.

Considering that the implementation of `torch.kthvalue` and `torch.Tensor.kthvalue` should be the same, this crash problem on both APIs should also be fixed by one pr before 1.11.0 version.

```
import torch

input = torch.rand([1, 210120], dtype=torch.float32)

def test():
    torch.Tensor.kthvalue(input.clone(), 9, dim=0)
test()
```
